web-app/src/components/HolidayModal/index.js

```diff
@@ -61,23 +72,48 @@ const HolidayModal = ({
           </Stat>
         </StatWrap>
         {shouldShowAdminControls() && (
-          <ButtonWrap>
-            <Button label="Approve" onClick={approveHoliday} />
-            <Button label="Reject" onClick={rejectHoliday} />
-          </ButtonWrap>
+          toggled === false ? <Stat>
+            <ButtonWrap>
+              <Button label="Approve" onClick={approveHoliday} />
+              <Button label="Reject" onClick={ expandRejectHolidayExplanationText } />
+            </ButtonWrap>
+          </Stat> : null
         )}
+        { toggled && ( 
+          <div>
+            <StatWrap>
+              <Stat>
+                <h2>{ 'Rejection Reason' }</h2>
+                <InputText onChange={assignRejectionReasonText}/>
+              </Stat>
+            </StatWrap>
+            <StatWrap>
+              <ButtonWrap>
+                <Button title={disableRejectionReasonButton ? 'Enter Message First' : ''} disabled={disableRejectionReasonButton} label="Confirm Rejection" onClick={() => rejectHoliday()} />
+              </ButtonWrap>
+            </StatWrap>
```

Comment:
These aren't stats, so if the StatWrap and Stat styling does transfer to here as well, then I feel it should be renamed. Or create another styled component.